### PR TITLE
Improve the parsing of methods in MySQL,fix issue #31547

### LINKED ISSUE #31547: Improve the parsing of methods in MySQL
## Background

Hi community.\
The ShardingSphere SQL parser engine helps users to parse SQL to create the AST (Abstract Syntax Tree) and visit the AST to get SQLStatement (Java Object). 

Currently, we are planning to enhance the support for MySQL SQL parsing in ShardingSphere.

More details:
https://shardingsphere.apache.org/document/current/en/reference/sharding/parse/

## Issue Background Explanation
In the original parsing work, it may have overlooked the parsing of method parameters. For ShardingSphere, we need to pay attention to the table name or field name in method parameters. If there are issues with parsing method parameters, it will cause problems in subsequent binding and rewriting tasks.

For the verification work, we need to complete the following items.
1. Find the example SQL of this method in the official website.[Built-In Function and Operator Reference](https://dev.mysql.com/doc/refman/8.0/en/built-in-function-reference.html)
2. verify if the SQL itself can be parsed by the parser
3. Check if the parsed SQLStatement can correctly capture the parameters in the method.

## Task

```
BIN()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bin)

```
BIN_TO_UUID()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_bin-to-uuid)
```
BIT_AND()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_bit-and)
```
BIT_COUNT()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/bit-functions.html#function_bit-count)
```
BIT_LENGTH()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_bit-length)
```
BIT_OR()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_bit-or)
```
BIT_XOR()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_bit-xor)
```
CAST()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/cast-functions.html#function_cast)
```
CEIL()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_ceil)
```
CEILING()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_ceiling)
```
CHAR()
```
- [ ] [link](https://dev.mysql.com/doc/refman/8.0/en/string-functions.html#function_char)

## Overall Procedure

If you intend to participate in fixing this issue, please feel free to leave a comment below the issue. Community members will assign the issue accordingly.  

><u>For example, you can leave a comment like this: "Hi, please assign this issue to me. Thank you!"</u>

Once you have claimed the issue, please review the syntax of the SQL on the official website of the corresponding database. Execute the SQL on the respective database to ensure the correctness of the SQL syntax.

><u>You can check the corresponding source of each SQL case on the official database website by clicking on the link provided below each case.</u> 

Next, execute the problematic SQL cases mentioned above in the database (you can quickly start the                     corresponding database using the Docker image for that database, and then connect to it using 
a client you are  familiar with), to ensure that the SQL syntax itself is correct.  

## Fixing ANTLR Grammar Parsing Issue

Once you have confirmed the correctness of the SQL syntax, you can validate and fix the grammar parsing issue in ShardingSphere. 

>If you are using IntelliJ IDEA, you will need to install the ANTLR plugin before proceeding.

If it is an ANTLR parsing error message, try to repair the `.g4` file by comparing it with the official database syntax until the SQL can be correctly parsed by ANTLR.

When there is no error message in the ANTLR Preview window, it means that ANTLR can correctly parse the SQL.  

## Visitor problem fix

After ANTLR parses SQL into an abstract syntax tree, ShardingSphere will access the abstract syntax tree through Visitor and extract the required information.
If you need to extract Segments, you need to first execute:
 ```
 mvn -T 2C clean install -DskipTests
 ```  
 Under the shardingsphere-parser module to compile the entire parser module.\
Then rewrite the corresponding visit method in SQLStatementVisitorr as needed to extract the corresponding Segment.

## Add assertion test file  

After the above SQL parsing problem is repaired, the corresponding Test needs to be added. 
The steps are as follows:
1. Add the corresponding `sql-case` in the `sql/supported` directory.
2. Add case assertions in the case directory of the `shardingsphere-test-it-parser` module.
3.  Run `org.apache.shardingsphere.test.it.sql.parser.internal.InternalSQLParserIT`\
 After SQL Parser IT runs successfully, you can submit a PR.   

## Relevant Skills

1. Master JAVA language
2. Have a basic understanding of Antlr g4 file
3. Be familiar with Doris SQLs

### PR DESCRIPTION
**Pull request by AbnerHuang2**
Fixes #31547.

Changes proposed in this pull request:
  Improve the parsing of methods in MySQL

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [x] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [x] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/it/parser/src/main/resources/sql/supported/dml/select-special-function.xml [hunk 1]
```diff
@@ -58,4 +58,12 @@
...
     <sql-case id="select_approx_percentile_cont_function" value="SELECT DeptId,APPROX_PERCENTILE_CONT(0.10) WITHIN GROUP(ORDER BY Salary) AS 'P10',APPROX_PERCENTILE_CONT(0.90) WITHIN GROUP(ORDER BY Salary) AS 'P90' FROM tblEmployee GROUP BY DeptId" db-types="SQLServer" />
     <sql-case id="select_wm_concat_function_with_schema" value="SELECT TO_CHAR(WMSYS.WM_CONCAT(DISTINCT o.status)) FROM t_order o GROUP BY o.order_id" db-types="Oracle" />
+    <sql-case id="select_bin" value="SELECT BIN(12)" db-types="MySQL" />
+    <sql-case id="select_bin_uuid" value="SELECT BIN_TO_UUID(UUID_TO_BIN('6ccd780c-baba-1026-9564-5b8c656024db'))" db-types="MySQL" />
+    <sql-case id="select_bit_length" value="SELECT BIT_LENGTH(1)" db-types="MySQL" />
+    <sql-case id="select_bit_count" value="SELECT BIT_COUNT(1)" db-types="MySQL" />
+    <sql-case id="select_ceil" value="SELECT CEIL(1.1)" db-types="MySQL" />
+    <sql-case id="select_ceiling" value="SELECT CEILING(1.1)" db-types="MySQL" />
+
```
Comment (strongduanmu) [inline test/it/parser/src/main/resources/sql/supported/dml/select-special-function.xml:67]: Please remove useless blank line.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

**Comment by strongduanmu** [comment]
Hi @AbnerHuang2 , thank you for your contribution. Can you solve code conflict?

File: test/it/parser/src/main/resources/sql/supported/dml/select-special-function.xml (hunk 1 above)
Comment (AbnerHuang2) [reply test/it/parser/src/main/resources/sql/supported/dml/select-special-function.xml]: done

**Comment by AbnerHuang2** [comment]
> Hi @AbnerHuang2 , thank you for your contribution. Can you solve code conflict?

done.

**Comment by AbnerHuang2** [comment]
And I found that functions BIT_XOR, CAST and CHAR already exists tests in projects. So I ignored them in this commit.

**Review by strongduanmu — APPROVED** [review]
Looks great, merged.

**Pull request merged**